aspnetcore/security/authentication/configure-oidc-web-authentication.md

```diff
@@ -0,0 +1,317 @@
+    options.TokenValidationParameters.RoleClaimType = "roles";
+});
+```
+
+See the following code example: 
+
+https://github.com/damienbod/keycloak-backchannel/tree/main/RazorPagePar
+
+### Implementing Microsoft identity providers
+
+Microsoft has multiple identity providers and OpenID Connect implementations. Microsoft has different OpenID Connect servers:
+
+* Microsoft Entra ID
+* Microsoft Entra External ID
+* Azure AD B2C
+
+If authenticating using one of the Microsoft identity providers in ASP.NET Core, it is recommended to use the [Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) Nuget packages.
+
+The Microsoft.Identity.Web Nuget packages is a Microsoft specific client built on top on the ASP.NET Core OpenID Connect client with some changes to the default client.
+
+## Using third party OpenID Connect provider clients
+
+Many OpenID Connect server implementations create Nuget packages which are optimized for the same OpenID Connect implementation. These packages implement the OpenID Connect client specifics with the extras required by the specific OpenID Connect server. Microsoft.Identity.Web is one example of this. 
+
+If implementing multiple OpenID Connect clients from different OpenID Connect servers in a single application, it is normally better to revert to the default ASP.NET Core implementation as the different clients overwrite some options which affect the other clients. 
+
+[OpenIddict Web providers](https://documentation.openiddict.com/integrations/web-providers) is a client implementation which supports many different server implementations.
+
+[IdentityModel](https://github.com/IdentityModel/IdentityModel) is a .NET standard helper library for claims-based identity, OAuth 2.0 and OpenID Connect. This can also be used to help with the client implementation.
+
+## Backend for frontend (BFF) security architecture
+
+It is no longer recommended to implement OpenID Connect public clients for any web applications. 
+
+See the [draft OAuth 2.0 for Browser-Based Applications](https://datatracker.ietf.org/doc/draft-ietf-oauth-browser-based-apps/) for further details.
+
+If implementing **web** applications which have no backend, it is recommended to use the [Backend for Frontend (BFF) pattern](/azure/architecture/patterns/backends-for-frontends) security architecture.  This pattern can be implemented in different ways, but the authentication is always implemented in the backend and no sensitive data is sent to the web client for further authorization or authentication flows.
+
+## Advanced features, standards, extending the OIDC client
+
+### Logging
+
+Debugging OpenID Connect clients can be hard. Personally identifiable information (PII) data is not logged by default. If debugging in development mode, the ** IdentityModelEventSource.ShowPII** can be used to log sensitive personal data. This should never by deployed to productive servers.
```

Comment:
This is a good callout. Can we also link to https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-8.0#configure-logging and mention that you may want to lower the configured log level to see everything?